packages/core/src/Editor.ts

```diff
@@ -276,7 +284,33 @@ export class Editor extends EventEmitter<EditorEvents> {
    * Creates a ProseMirror view.
    */
   private createView(): void {
-    const doc = createDocument(this.options.content, this.schema, this.options.parseOptions)
+    let doc: ProseMirrorNode
+
+    try {
+      doc = createDocument(
+        this.options.content,
+        this.schema,
+        this.options.parseOptions,
+        { errorOnInvalidContent: this.options.enableContentCheck },
+      )
+    } catch (e) {
+
+      // Remove the collaboration extension if the content is invalid, to not sync invalid content
+      this.options.extensions = this.options.extensions.filter(extension => extension.name !== 'collaboration')
+
+      // Recreate the extension manager without the invalid extensions
+      this.createExtensionManager()
+
+      this.emit('contentError', { editor: this, error: e as Error })
+
+      // Content is invalid, but attempt to create it anyway, stripping out the invalid parts
```

Comment:
Above, we prevent a sync to other collaborative users, but regardless of this, the doc is changed and `onUpdate` is triggered. In a setup without collaboration, as a developer, I would have to hook in accordingly when saving (e.g. through the changed doc or onUpdate) and check whether an error was previously emitted. Is this correct?

Reply:
Awesome. Thank you.